stdlib/Random/docs/src/index.md

```diff
@@ -45,6 +53,222 @@ Random.shuffle
 Random.shuffle!
 ```
 
+## Generators (creation and seeding)
+
+```@docs
+Random.srand
+Random.MersenneTwister
+Random.RandomDevice
+```
+
+## Hooking into the `Random` API
+
+There are two mostly orthogonal ways to extend `Random` functionalities:
+1) generating random values of custom types
+2) creating new generators
+
+The API for 1) is quite functional, but is relatively recent so it may still have to evolve in subsequent releases of the `Random` module.
+For example, it's typically sufficient to implement one `rand` method in order to have all other usual methods work automatically.
+
+The API for 2) is still rudimentary, and may require more work than strictly necessary from the implementor,
+in order to support usual types of generated values.
+
+### Generating random values of custom types
+
+There are two categories: generating values from a type (e.g. `rand(Int)`), or from a collection (e.g. `rand(1:3)`).
+The simple cases are explained first, and more advanced usage is presented later.
+We assume here that the choice of algorithm is independent of the RNG, so we use `AbstractRNG` in our signatures.
+
+#### Generating values from a type
+
+Given a type `T`, it's currently assumed that if `rand(T)` is defined, an object of type `T` will be produced.
+In order to define random generation of values of type `T`, the following method can be defined:
+`rand(rng::AbstractRNG, ::Random.SamplerType{T})` (this should return what `rand(rng, T)` is expected to return).
+
+Let's take the following example: we implement a `Die` type, with a variable number `n` of sides, numbered from `1` to `n`.
+We want `rand(Die)` to produce a die with a random number of up to 20 sides (and at least 4):
+
+```julia
+struct Die
+    nsides::Int # number of sides
+end
+
```

Comment:
IMO we should  add 
```
using Random
```
to prevent deprecations warning

Reply:
I think a bunch of stdlib modules assume an implicit corresponding `using` directive...